### DISCUSSION TITLE
[question] Can meep compute static magnetic field?

### DISCUSSION THREAD

**Discussion by yurivict** [NanoComp/meep#2892] [Q&A]
Can meep compute static magnetic field based on the 3D distributions of electric current and magnetic materials, for example iron, neodymium ?

**Comment by stevengj** [comment]
In principle, you could just turn on a static current distribution and wait for it to reach steady state.   However, it doesn't implement ferromagnetic materials.

In practice, I wouldn't use a time-domain program to do magneto-static calculations.  You're generally going to be much better off using an algorithm designed for statics, since it boils down to much simpler Laplace-type equations.

**Comment by yurivict** [comment]
> You're generally going to be much better off using an algorithm designed for statics [...]

In a lot of situations electromagnetic waves are also essentially a static problem, if to factor out the constant oscillations.
Doesn't meep already contain solvers for such problems without the time component?

Static magnetic field problems aren't much different.
